lib/src/rules/prefer_final_fields.dart

```diff
@@ -97,21 +82,38 @@ class PreferFinalFields extends LintRule {
 
 class _Visitor extends SimpleAstVisitor {
   final LintRule rule;
+  final Set<Element> _mutatedElements = new HashSet<Element>();
```

Comment:
It would be good to clear this set after the visit to avoid holding on to memory we no longer need. One possibility would be to not cache the visitor in the LintRule (I think the getter is only called once, so we could re-create the visitor on each call).